docs/source/posts/2024/2024-06-06-week-1-robin.rst

```diff
@@ -0,0 +1,106 @@
+
+why phi-3-mini-4k-instruct?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before I detail everything I've done this week, I'll explain why `phi-3 mini 4k <https://huggingface.co/microsoft/Phi-3-mini-4k-instruct>`_ was chosen as the LLM, I forgot to mention this in the last blog. Phi-3 is a small 3.8B 4k context model. Due to its small size, it runs fast both locally and on huggingface. Performance wise comparatively with other opensource models, it performs decently well. In the `LMYSYS LLM leaderboard <https://chat.lmsys.org/?leaderboard>`_ phi-3 mini 4k comes with an ELO of 1066 (59th position). But it achieves this as a small model.
+I also tried Llama3-8B, it performs better than phi-3 mini with an ELO of 1153 and rank 22. But it is considerably slower for inference. Due to this, I chose phi-3 mini for now.
+
+
+Things I did week-1 (and some week2)
+------------------------------------
+
+1) **Choosing the vector database**
+
+I decided to choose `Pinecone <https://www.pinecone.io/>`_ as the vector DB because it had a very generous free tier. Other options on consideration were `pgvector <https://github.com/pgvector/pgvector>`_ and `chromadb <https://www.trychroma.com/>`_, but they didn't have a free tier.
+
+2) **PR Submissions and Review**
+
+I also merged a `PR <https://github.com/fury-gl/fury/pull/891>`_ on FURY which fixes a CI issue. I also spent time doing review of other PRs from my fellow GSoC mates.
+
+3) **Deciding which embedding model to use**
+
+A good embedding model is necessary to generate embeddings which we then upsert into the DB. Ollama had embedding model support, but I found the catalogue very small and the models they provided were not powerful enough. Therefore I decided to try using HuggingFace Sentence Transformers.
+Sentence Transformers have a very vibrant catalogue of models available of various sizes. I chose `gte-large-en-v1.5 <https://huggingface.co/Alibaba-NLP/gte-large-en-v1.5>`_ from Alibaba-NLP, an 8k context, 434 million parameter model. It only had a modest memory requirement of 1.62 GB.
+Performance wise, it ranks 11th on the `MTEB leaderboard <https://huggingface.co/spaces/mteb/leaderboard>`_. It is a very interesting model due to its size:performance ratio.
+
+4) **Hosting the embedding model**
+
+Hosting this sentence-transformer model was confusing. For some reason, the HF spaces were blocking the Python script from writing on ``.cache`` folder. Later I figured out the reason and fixed it.
+I've hosted 5 gunicorn workers to serve 5 parallel requests at a time. Since the model is small, this is possible.
+
+5) **Hosting the database endpoint**
+
+I wrapped the pinecone db api into an endpoint so it'll be easy to query and receive the results.
```

Comment:
When you write acronyms, make sure they are in uppercase, it improves readability.
`db -> DB`
`api -> API`